source/Engine/Bytecode/StandardLibrary.cpp

```diff
@@ -4416,6 +4416,150 @@ VMValue Draw_Glyph(int argCount, VMValue* args, Uint32 threadID) {
 
 	return NULL_VAL;
 }
+/***
+ * Draw.TextArray
+ * \desc Draws a series of sprites based on a converted sprite string.
+ * \param sprite (Integer): The index of the loaded sprite to be used as text.
+ * \param animation (Integer): The animation index.
+ * \param x (Number): The X value to begin drawing.
+ * \param y (Number): The Y value to begin drawing.
+ * \param string (Array): The array containing frame indexes.
+ * \param startFrame (Integer): The index to begin drawing.
+ * \param endFrame (Integer): The index to end drawing.
+ * \param align (Integer): The text alignment.
+ * \param spacing (Integer): The space between drawn sprites.
+ * \paramOpt charOffsetsX (Array): The X offsets at which to draw per frame. Must also have charOffsetsY to be used.
+ * \paramOpt charOffsetsY (Array): The Y offsets at which to draw per frame.
+ * \ns Draw
+ */
+VMValue Draw_TextArray(int argCount, VMValue* args, Uint32 threadID) {
+	CHECK_AT_LEAST_ARGCOUNT(9);
+	if (ScriptManager::Lock()) {
+		ISprite* sprite = GET_ARG(0, GetSprite);
+		int animation = GET_ARG(1, GetInteger);
+		float x = GET_ARG(2, GetDecimal);
+		float y = GET_ARG(3, GetDecimal);
+		ObjArray* string = GET_ARG(4, GetArray);
+		int startFrame = GET_ARG(5, GetInteger);
+		int endFrame = GET_ARG(6, GetInteger);
+		int align = GET_ARG(7, GetInteger);
+		int spacing = GET_ARG(8, GetInteger);
+		ObjArray* charOffsetsX = GET_ARG_OPT(9, GetArray, NewArray());
+		ObjArray* charOffsetsY = GET_ARG_OPT(10, GetArray, NewArray());
```

Comment:
This needlessly creates empty arrays every time this method is called without those arguments, which can strain the GC.